Sources/Layout/MessageSizeCalculator.swift

```diff
@@ -215,9 +232,14 @@ open class MessageSizeCalculator: CellSizeCalculator {
 
     internal func labelSize(for attributedText: NSAttributedString, considering maxWidth: CGFloat) -> CGSize {
         let constraintBox = CGSize(width: maxWidth, height: .greatestFiniteMagnitude)
-        let rect = attributedText.boundingRect(with: constraintBox, options: [.usesLineFragmentOrigin, .usesFontLeading], context: nil).integral
-
-        return rect.size
+        
+        textContainer.size = constraintBox
+        textStorage.replaceCharacters(in: NSRange(location: 0, length: textStorage.length), with: attributedText)
```

Comment:
Currently, we have no thread-safe issue because we do this on main thread. So we can just replace characters.